ginkgo/ginkgo.go

```diff
@@ -0,0 +1,362 @@
+
+	return ginkgo.XDescribe(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("XDescribe", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func Context(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.Context(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("Context", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func FContext(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.FContext(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("FContext", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func PContext(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.PContext(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("PContext", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func XContext(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.XContext(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("XContext", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func When(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.When(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("When", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func FWhen(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.FWhen(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("FWhen", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func PWhen(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.PWhen(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("PWhen", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+func XWhen(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.XWhen(text, utils.PatchArgsNoArgs(func() error { return currentSuite.EnterContainer("XWhen", text) }, currentSuite.LeaveContainer, callback...)...)
+}
+
+/******** Leafs **********/
+
+func It(text string, callback ...interface{}) bool {
+	initSetup()
+
+	test := currentSuite.Snapshot().RegisterTest()
+
+	suiteName := utils.GetSuiteName()
+
+	return ginkgo.It(text, utils.PatchArgsNoArgs(func() error { return test.Enter("It", suiteName) }, test.Leave, callback...)...)
+}
+
+func FIt(text string, callback ...interface{}) bool {
+	initSetup()
+
+	test := currentSuite.Snapshot().RegisterTest()
+
+	suiteName := utils.GetSuiteName()
+
+	return ginkgo.FIt(text, utils.PatchArgsNoArgs(func() error { return test.Enter("FIt", suiteName) }, test.Leave, callback...)...)
+}
+
+func PIt(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.PIt(text, callback...)
+}
+
+func XIt(text string, callback ...interface{}) bool {
+	initSetup()
+
+	return ginkgo.XIt(text, callback...)
+}
```

Comment:
Don't we want to create spans on these two cases? If it is intentional, can you add a comment explaining why?

Reply:
You are right I should explain why. What about the following comment:

```golang
// no need to register this test since it will be ignored
// see pending spec: https://onsi.github.io/ginkgo/#pending-specs
```

Reply:
LGTM